llama-index-core/llama_index/core/tools/function_tool.py

```diff
@@ -30,11 +40,16 @@ class FunctionTool(AsyncBaseTool):
 
     def __init__(
         self,
-        fn: Callable[..., Any],
         metadata: ToolMetadata,
+        fn: Optional[Callable[..., Any]] = None,
```

Comment:
A little scary to be making a breaking change to the init interface 👀

Reply:
Maybe we can do this instead?
```python
def __init__(
  self,
  fn: Optional[Callable[..., Any]],
  metadata: ToolMetadata,
  async_fn: Optional[AsyncCallable],
) -> None:
  <handling for fn and async fn>
```

Reply:
Ok, made a small tweak. Its not great, but we've made a lot of breaking changes lately so I didn't want to pile on :)